src/streaming/utils/TTMLParser.js

```diff
@@ -71,6 +73,133 @@ function TTMLParser() {
     }
 
     /**
+     * Get the begin-end interval if present, or null otherwise.
+     *
+     * @param {Object} element - TTML element which may have begin and end attributes
+     */
+    function getInterval(element) {
+        if (element.hasOwnProperty('begin') && element.hasOwnProperty('end')) {
```

Comment:
This test for presence of both a begin and an end time being required could easily be relaxed if the code can access the begin and end time of the sample: a missing begin time can be replaced by the begin time of the sample; a missing end by the end time of the sample.